### PR TITLE
Cleanup: Remove questionable syntax

### PR DESCRIPTION
**Pull request by Eddi-z**

### CODE REVIEW

**Comment by nielsmh** [comment]
That's some amazingly bad syntax. Is that from the dawn of time?

**Comment by Eddi-z** [comment]
i'm not 100% convinced that the track one is really the best/clearest approach to this

> That's some amazingly bad syntax. Is that from the dawn of time?

i'm fairly sure that's from r1.
and it probably makes perfect sense if you're reading assembler code and translating that into C code. but if you're reading this with an "if X then do Y" mindset, the conditions are all backwards and confusing.

**Review by M3Henry — CHANGES_REQUESTED** [review]
This looks like mixed code and data. I would use an array of score-boundaries and use std::lower_bound to figure out the offset of val.

**Comment by nielsmh** [comment]
Yep, this dates back to r1:
https://github.com/OpenTTD/OpenTTD/blame/2c27f45cdc31c0466d72eb8d1539448b93af0d5b/economy.c#L56-L60

**Comment by Eddi-z** [comment]
skipped the track one for now, as probably a more involved rewrite might be in order there

File: src/order_gui.cpp [hunk 1]
```diff
@@ -395,11 +395,19 @@ static Order GetOrderCmdFromTile(const Vehicle *v, TileIndex tile)
...
-			(facil = FACIL_TRAIN, v->type == VEH_TRAIN) ||
-			(facil = FACIL_AIRPORT, v->type == VEH_AIRCRAFT) ||
-			(facil = FACIL_BUS_STOP, v->type == VEH_ROAD && RoadVehicle::From(v)->IsBus()) ||
-			(facil = FACIL_TRUCK_STOP, 1);
+			switch (v->type) {
+				case VEH_SHIP:     facil = FACIL_DOCK;    break;
+				case VEH_TRAIN:    facil = FACIL_TRAIN;   break;
+				case VEH_AIRCRAFT: facil = FACIL_AIRPORT; break;
+				case VEH_ROAD:
+					if (RoadVehicle::From(v)->IsBus()) {
```
Comment (LordAro) [inline src/order_gui.cpp]: wouldn't be awful if this was turned into a ternary, imo

File: src/rail_cmd.cpp [hunk 2]
```diff
@@ -2285,23 +2285,32 @@ static void DrawTrackBits(TileInfo *ti, TrackBits track)
...
-			(image = rti->base_sprites.track_ns, track == TRACK_BIT_HORZ) ||
-			(image++,                            track == TRACK_BIT_VERT) ||
-
-			(junction = true, false) ||
-			(image = rti->base_sprites.ground, (track & TRACK_BIT_3WAY_NE) == 0) ||
-			(image++,                          (track & TRACK_BIT_3WAY_SW) == 0) ||
-			(image++,                          (track & TRACK_BIT_3WAY_NW) == 0) ||
-			(image++,                          (track & TRACK_BIT_3WAY_SE) == 0) ||
-			(image++, true);
+
```
Comment (PeterN) [inline src/rail_cmd.cpp]: Don't need the blank line

File: src/rail_cmd.cpp [hunk 3]
```diff
@@ -2285,23 +2285,32 @@ static void DrawTrackBits(TileInfo *ti, TrackBits track)
...
+				default:
+					junction = true;
+					if ((track & TRACK_BIT_3WAY_NE) == 0) { image = rti->base_sprites.ground;     break; }
+					if ((track & TRACK_BIT_3WAY_SW) == 0) { image = rti->base_sprites.ground + 1; break; }
+					if ((track & TRACK_BIT_3WAY_NW) == 0) { image = rti->base_sprites.ground + 2; break; }
+					if ((track & TRACK_BIT_3WAY_SE) == 0) { image = rti->base_sprites.ground + 3; break; }
+					image = rti->base_sprites.ground + 4;
+					break;
+			}
+
```
Comment (PeterN) [inline src/rail_cmd.cpp]: Nor this blank line

**Review by PeterN — APPROVED** [review]

**Pull request merged**